src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -76,16 +76,42 @@
         (rr/status ((response :outcome) outcomes->codes))
         (rr/content-type "text/plain"))))
 
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;;; Web app
-
 (defn try-to-parse
   [body]
   (try
     (cheshire/parse-stream (io/reader body) true)
     (catch Exception e
       (log/debug e))))
 
+(schema/defn handle-cert-clean
+  [{:keys [body]}
+   ca-settings :- ca/CaSettings]
+  (let [{:keys [certnames async]} (try-to-parse body)
+        [existing-certs missing-certs] (split-with
+                                        #(ca/certificate-exists? ca-settings %)
+                                        certnames)
+        message (when (seq missing-certs)
+                  (format "The following certs do not exist and cannot be revoked: %s"
+                          (vec missing-certs)))]
+    ;; TODO support async mode
+    (if (true? async)
```

Comment:
Is it worth having this check right at the beginning, so we don't bother with checking cert existence if we're just gonna fail?

Reply:
I can try that.